docs/guides/javascript.rst

```diff
@@ -12,88 +12,159 @@ The 1.8 version is still functional and is :ref:`described below<1.8-js>`.
 AMD
 ===
 
-Defining and loading a module in Elgg 1.9 takes two steps:
+Here we'll describe executing AMD modules and how to make them.
 
-1. Define your module as asynchronous JavaScript.
-2. Tell Elgg to asynchronously execute your module in the current page.
+Executing a module in the current page
+======================================
 
-1. Define your module as asynchronous JavaScript
-------------------------------------------------
+Once a module is defined, telling Elgg to load it in the current page is easy:
 
-You can define a module by creating a view or registering a URL.
+.. code-block:: php
+
+    <?php
+    elgg_require_js("myplugin/say_hello");
+
+On the client-side, this will asynchronously load the module, load any dependencies, and
+execute it.
+
+This can also be done from JavaScript by calling ``require(["myplugin/say_hello"])``,
+but you should not need to.
+
+Defining the Module
+===================
 
-Defining modules as a view
-^^^^^^^^^^^^^^^^^^^^^^^^^^
+A module is essentially a list of dependencies and a function which is executed, maybe
+returning a value to other modules. It can perform operations on the page, return an API
+or value for consumption by other modules, or both.
 
-Modules defined by creating views are immediately available for use and require no registration.
-To register a module named ``my/module``, create the view ``views/default/js/my/module.js``.
+Here we define a basic module ``myplugin/say_hello`` for altering the page:
+
+.. code-block:: javascript
+
+    // in views/default/js/myplugin/say_hello.js
+
+    define(function(require) {
+        var elgg = require("elgg");
+        var $ = require("jquery");
+
+        $('body').append(elgg.echo('hello_world'));
+    });
+
+.. note::
+
+    RequireJS uses a bit of magic to recognize the ``elgg`` and ``jquery`` modules as
+    dependencies, even before executing the function!
```

Comment:
This is unnecessary for the level of user we're addressing. Maybe more important to mention that the "require" argument is *required.* It will not work if you just write `define(function() { require(...) });`